From 41aefec0b6d678ac0bd39e18a6277c8e39087339 Mon Sep 17 00:00:00 2001
From: Wang Hai <wanghai38@huawei.com>
Date: Mon, 22 Aug 2022 20:03:59 +0800
Subject: [PATCH 023/261] [Huawei] thermal/core: Revert "[Backport]
 thermal/core: Fix memory leak in the error path"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187506

--------------------------------

This reverts commit 7a539db5fa285a00677b80ca5ae3e0ebc900af21.

commit 5674d780b865 ("[Backport] thermal/drivers/core: Use a char pointer
for the cooling device name") will be reverted. This is its fix patch, so
it also needs to be revert.

Signed-off-by: Wang Hai <wanghai38@huawei.com>
---
 drivers/thermal/thermal_core.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/thermal/thermal_core.c b/drivers/thermal/thermal_core.c
index dd449945e1e5..c0d8c882b247 100644
--- a/drivers/thermal/thermal_core.c
+++ b/drivers/thermal/thermal_core.c
@@ -1152,7 +1152,6 @@ __thermal_cooling_device_register(struct device_node *np,
 out_ida_remove:
 	ida_simple_remove(&thermal_cdev_ida, id);
 out_kfree_cdev:
-	kfree(cdev);
 	return ERR_PTR(ret);
 }
 
-- 
2.17.1

